components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -100,26 +100,55 @@ void BlockchainRegistry::GetAllTokens(const std::string& chain_id,
   std::move(callback).Run(std::move(tokens_copy));
 }
 
-void BlockchainRegistry::GetBuyTokens(const std::string& chain_id,
+void BlockchainRegistry::GetBuyTokens(mojom::OnRampProvider provider,
+                                      const std::string& chain_id,
                                       GetBuyTokensCallback callback) {
   std::vector<brave_wallet::mojom::BlockchainTokenPtr> blockchain_buy_tokens;
-  for (auto token : *kBuyTokens) {
+  std::vector<mojom::BlockchainToken> buy_tokens;
+  if (provider == mojom::OnRampProvider::Wyre)
+    buy_tokens = *kWyreBuyTokens;
+  else if (provider == mojom::OnRampProvider::Ramp)
+    buy_tokens = *kRampBuyTokens;
+
+  for (auto token : buy_tokens) {
+    if (token.chain_id != chain_id) {
+      continue;
+    }
+
     auto blockchain_token = brave_wallet::mojom::BlockchainToken::New();
     *blockchain_token = token;
     blockchain_buy_tokens.push_back(std::move(blockchain_token));
   }
   std::move(callback).Run(std::move(blockchain_buy_tokens));
 }
 
-void BlockchainRegistry::GetBuyUrl(const std::string& chain_id,
+void BlockchainRegistry::GetBuyUrl(mojom::OnRampProvider provider,
+                                   const std::string& chain_id,
                                    const std::string& address,
                                    const std::string& symbol,
                                    const std::string& amount,
                                    GetBuyUrlCallback callback) {
-  std::string url = base::StringPrintf(kBuyUrl, address.c_str(), symbol.c_str(),
-                                       amount.c_str(), kWyreID);
-
-  std::move(callback).Run(url);
+  std::string url = "";
+  if (provider == mojom::OnRampProvider::Wyre) {
+    if (chain_id != mojom::kMainnetChainId) {
+      std::move(callback).Run(url, "UNSUPPORTED_NETWORK");
+      return;
+    }
+
+    url = base::StringPrintf(kWyreBuyUrl, address.c_str(), symbol.c_str(),
+                             amount.c_str(), kWyreID);
+    std::move(callback).Run(url, absl::nullopt);
+  } else if (provider == mojom::OnRampProvider::Ramp) {
+    if (chain_id != mojom::kMainnetChainId) {
+      std::move(callback).Run(url, "UNSUPPORTED_NETWORK");
+      return;
+    }
+
+    url = base::StringPrintf(kRampBuyUrl, address.c_str(), symbol.c_str(),
+                             amount.c_str());
+    std::move(callback).Run(url, absl::nullopt);
```

Comment:
`std::move(url)` here too

Reply:
Fixed here: fdfd2f5b0d086b4d433080c750993185e3f35f21.

By the way, do we also need an `std::move(url)` when url is an empty string (ex, while handling errors)?

Reply:
Not moving an empty string is OK. Not moving non-empty string may cause copying.